storm-core/src/jvm/backtype/storm/grouping/PartialKeyGrouping.java

```diff
@@ -65,7 +66,11 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
                 List<Object> selectedFields = outFields.select(fields, values);
                 ByteBuffer out = ByteBuffer.allocate(selectedFields.size() * 4);
                 for (Object o: selectedFields) {
-                    out.putInt(o.hashCode());
+                    if (o instanceof Object[]) {
```

Comment:
@d2r 
I didn't think about primitive array type.
Actually I mean that,
- Does 'o' have an chance to be instance of List? 
- If then, could we apply ```((List)o).toArray()``` to convert it Object[], and let if statement handle it?